tools/modules/update.sh

```diff
@@ -16,28 +16,30 @@
 # update_modules.sh updates the repositories for external code (to the precise
 # revisions used for testing).
 #
-# If the single argument `--git_only` is passed, the repositories will be
+# If the single argument `--fetch_only` is passed, the repositories will be
 # updated to the pinned versions without configuring or building them.
 # If the argument `--build_only` is passed, the script will assume that the
 # repositories are at the correct version and will configure and build them.
 
-git_maybe_clone() {
-  local repo="$1"
+wget_copy_archive() {
+  local target="$1"
   local dir="$2"
-  if [[ ! -d "$dir/.git" ]]; then
-    git clone "$repo" "$dir"
-  fi
-}
+  local sha="$3"
 
-git_checkout_sha() {
-  local repo="$1"
-  local sha="$2"
-  cd "$repo"
-  if [[ "$sha" != "$(git rev-parse HEAD)" ]]; then
-    git fetch origin master
-    git checkout -f "$sha"
+  if [[ ! -f "$dir/$sha.sentinel" ]]; then
+    wget "https://github.com/llvm-mirror/$target/archive/$sha.zip"
+    rm -rf "$dir"
+    mkdir -p "$dir"
+    local tmpdir=$(mktemp -d)
```

Comment:
Please add a `trap` to cleanup the temporary directory: `trap 'rm "$tmpdir"' EXIT ERR INT`

Reply:
:+1:

Also above for the .zip file while we're at it.

Reply:
actually, to make this even more robust I am going to cd back to the $ROOT, since some parts later on will cd away.  So depending on where it dies, it'll be safer to always cd back to known $ROOT and rm from there.

(Also I just learned the hard way that purposefully interrupting tools/modules/update.sh in the middle and leaving clang/llvm in a bad state makes the presubmit very unhappy)